### PR TITLE
ch4/gpu: Add CVAR for choosing GPU IPC protocol

### PR DESCRIPTION
**Pull request by abrooks98**
## Pull Request Description

This PR adds support for selecting the IPC protocol (i.e. read vs. write) and the GPU engine type for IPC paths with an option for auto-selection based on the environment and scenario.

Additionally this PR adds a fix to cleanup `mask_contents` in the ZE MPL backend

## Author Checklist
* [x] **Provide Description** 
      Particularly focus on _why_, not _what_. Reference background, issues, test failures, xfail entries, etc.
* [x] **Commits Follow Good Practice**
      Commits are self-contained and do not do two things at once. 
      Commit message is of the form: `module: short description` 
      Commit message explains what's in the commit.
* [ ] **Passes All Tests**
      Whitespace checker. Warnings test. Additional tests via comments.
* [x] **Contribution Agreement**
      For non-Argonne authors, check [contribution agreement](http://www.mpich.org/documentation/contributor-docs/). 
      If necessary, request an explicit comment from your companies PR approval manager.

### CODE REVIEW

**Comment by abrooks98** [comment]
test:mpich/ch4/gpu

File: src/mpl/src/gpu/mpl_gpu_ze.c [hunk 1]
```diff
@@ -1271,6 +1271,9 @@ int MPL_gpu_finalize(void)
 
     MPL_event_pool_destroy();
 
+    MPL_free(mask_contents.dev_id);
+    MPL_free(mask_contents.subdev_id);
```
Comment (hzhou) [inline src/mpl/src/gpu/mpl_gpu_ze.c:1275]: Were these leaks caught by the gpu-debug tests?

**Review by hzhou — APPROVED** [review]
LGTM. Just a question out-of-curiosity.

File: src/mpl/src/gpu/mpl_gpu_ze.c (hunk 1 above)
Comment (abrooks98) [reply src/mpl/src/gpu/mpl_gpu_ze.c:1275]: Yeah we caught these when internally running the debug GPU tests for https://github.com/pmodels/mpich/pull/6451

**Review by raffenet — APPROVED** [review]

**Comment by abrooks98** [comment]
ofi-gpu tests are clean; rebasing in preparation for merge

**Pull request merged**